App/features/group/rename-group/sagas.ts

```diff
@@ -0,0 +1,22 @@
+import { ActionType, getType } from 'typesafe-actions'
+import { call, put, takeEvery, all } from 'redux-saga/effects'
+import Textile from '@textile/react-native-sdk'
+import { renameGroup } from './actions'
+import PhotoViewingActions from '../../../Redux/PhotoViewingRedux'
+
+export function *handleRenameGroupRequest(action: ActionType<typeof renameGroup.request>) {
+  const { threadId, name } = action.payload
+  try {
+    yield call(Textile.threads.rename, threadId, name)
+    yield put(PhotoViewingActions.updateThreadName(threadId, name))
```

Comment:
We have this situation now where the list of threads is in a different place in the redux tree than the group data. You can see the consequence of that here where you have to dispatch this random action for another reducer. I've been trying to think of a nice way to consolidate these two sources of information. Will be good to figure out soon.

Reply:
I think it'd be ideal to just have a single threads reducer that stores all thread/group related data-plus refactoring the two branches of the redux tree together would give us the opportunity to normalize some stuff like using `createAsyncAction`.